src/extended_json.ts

```diff
@@ -362,13 +370,18 @@ function serializeDocument(doc: any, options: EJSONSerializeOptions) {
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function parse(text: string, options?: EJSONOptions): any {
+  options = {
```

Comment:
LGTM, although I think we've preferred to create a new variable rather than overriding the options argument variable. ex. `const ejsonOptions = ...`